src/Common/src/Interop/Unix/System.Net.Http.Native/Interop.SList.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class LibCurl
+    {
+        [DllImport(Libraries.HttpNative, CharSet = CharSet.Ansi)]
+        private static extern IntPtr SListAppend(IntPtr slist, string headerValue);
+
+        internal static bool SListAppend(SafeCurlSlistHandle slist, string headerValue)
+        {
+            bool gotRef = false;
+            try
+            {
+                slist.DangerousAddRef(ref gotRef);
+                IntPtr newHandle = SListAppend(slist.DangerousGetHandle(), headerValue);
+                if (newHandle != IntPtr.Zero)
+                {
+                    slist.SetHandle(newHandle);
+                    return true;
+                }
+                return false;
+            }
+            finally
+            {
+                if (!gotRef)
+                    slist.DangerousRelease();
+            }
+        }
+
+        [DllImport(Libraries.HttpNative)]
+        private static extern void SListFreeAll(IntPtr slist);
+
+        internal sealed class SafeCurlSlistHandle : SafeHandle
```

Comment:
In the name of the functions we've used ```SList```, but here we're using ```Slist```... we should be consistent, probably with the former.